sdk/storage/azure-storage-file-datalake/README.md

```diff
@@ -0,0 +1,199 @@
+    It provides operations to retrieve and configure the account properties
+    as well as list, create, and delete file systems within the account.
+    For operations relating to a specific file system, directory or file, clients for those entities
+    can also be retrieved using the `get_file_client`, `get_directory_client` or `get_file_system_client` functions.
+2. **FileSystemClient** - this client represents interaction with a specific
+    file system, even if that file system need not exist yet. It provides operations to create, delete, or
+    configure file systems and includes operations to list paths under file system, upload, and delete file or
+    directory in the file system.  
+    For operations relating to a specific file, the client can also be retrieved using
+    the `get_file_client` function.  
+    For operations relating to a specific directory, the client can be retrieved using 
+    the `get_directory_client` function.  
+3. **DataLakeDirectoryClient** - this client represents interaction with a specific
+    directory, even if that directory need not exist yet. It provides directory operations create, delete, rename,
+    get properties and set properties operations.
+3. **DataLakeFileClient** - this client represents interaction with a specific
+    file, even if that file need not exist yet. It provides file operations to append data, flush data, delete, 
+    create, and read file.
+4. **DataLakeLeaseClient** - this client represents lease interactions with a FileSystemClient, DataLakeDirectoryClient
+    or DataLakeFileClient. It provides operations to acquire, renew, release, change, and break leases on the resources.
+
+## Examples
+
+The following sections provide several code snippets covering some of the most common Storage DataLake tasks, including:
+
+* [Client creation with a connection string](#client-creation-with-a-connection-string)
+* [Uploading a file](#uploading-a-file)
+* [Downloading a file](#downloading-a-file)
+* [Enumerating paths](#enumerating-paths)
+
+
+### Client creation with a connection string
+Create the DataLakeServiceClient using the connection string to your Azure Storage account.
+
+```python
+from azure.storage.file.datalake import DataLakeServiceClient
+
+service = DataLakeServiceClient.from_connection_string(conn_str="my_connection_string")
+```
+
+### Uploading a file
+Upload a file to your file system.
+
+```python
+from azure.storage.file.datalake import DataLakeFileClient
+
+data = b"abc"
+file = DataLakeFileClient.from_connection_string("my_connection_string", 
+                                                 file_system_name="myfilesystem", file_path="myfile")
+
+file.append_data(data, offset=0, length=len(data))
+file.flush_data(len(data))
+```
+
+### Downloading a file
+Download a file from your file system.
+
+```python
+from azure.storage.file.datalake import DataLakeFileClient
```

Comment:
Did we decide to go with file.datalake or filedatalake?

Reply:
we didn't 😢